catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/TransformersCommand.java

```diff
@@ -52,95 +57,194 @@
-    public String printProperties() {
+    private String printDefaultProperties() {
+
+      String id = getTransformerPropertyString(Constants.SERVICE_ID);
+      String schema = getTransformerPropertyString(SCHEMA);
+      List<String> mimeTypes = getTransformerMimeTypes();
 
       StringBuilder s =
           new StringBuilder(
               MessageFormat.format(
                   "{0}: {1}\n\n\t{2}: {3}\n\t{4}s: {5}\n",
                   Constants.SERVICE_ID, id, SCHEMA, schema, MIME_TYPE, mimeTypes.remove(0)));
 
-      for (String mimeType : mimeTypes) {
-        s.append("\t\t    ").append(mimeType).append("\n");
+      s.append(printMimeTypes(mimeTypes));
+
+      return s.toString();
+    }
+
+    private String printAllProperties() {
+
+      StringBuilder s = new StringBuilder();
+      String[] propertyKeys = ref.getPropertyKeys();
+      List<String> mimeTypes = getTransformerMimeTypes();
+      String id = getTransformerPropertyString(Constants.SERVICE_ID);
+      String value;
+
+      s.append(MessageFormat.format("{0}: {1}\n\n\t", Constants.SERVICE_ID, id));
+
+      for (String key : propertyKeys) {
+        if (key.equals(MIME_TYPE)) {
+          continue;
+        }
+
+        if (key.equals(Constants.SERVICE_ID)) {
+          continue;
+        }
+
+        value = getTransformerPropertyString(key);
+        s.append(MessageFormat.format("\n\t{0}: {1}", key, value));
       }
+
+      s.append(MessageFormat.format("\n\t{0}: {1}\n", MIME_TYPE, mimeTypes.remove(0)));
+      s.append(printMimeTypes(mimeTypes));
       return s.toString();
     }
 
-    private String getTransformerPropertyString(ServiceReference ref, String property) {
+    private String getTransformerPropertyString(String property) {
 
       return Optional.ofNullable(ref.getProperty(property))
-          .map(String.class::cast)
-          .filter(String.class::isInstance)
+          .map(Object::toString)
           .orElse(NOT_AVAILABLE);
     }
 
-    private List<String> getTransformerMimeTypes(ServiceReference ref) {
+    private List<String> getTransformerMimeTypes() {
 
       List<String> mimeProperties;
 
       if (ref.getProperty(MIME_TYPE) instanceof List) {
         mimeProperties = (ArrayList<String>) ref.getProperty(MIME_TYPE);
       } else {
         mimeProperties = new ArrayList<>();
-        mimeProperties.add(getTransformerPropertyString(ref, MIME_TYPE));
+        mimeProperties.add(getTransformerPropertyString(MIME_TYPE));
       }
       return mimeProperties;
     }
+
+    private String printMimeTypes(List<String> mimeTypes) {
+      StringBuilder s = new StringBuilder();
+
+      for (String mimeType : mimeTypes) {
+        s.append("\t\t    ").append(mimeType).append("\n");
+      }
+
+      return s.toString();
+    }
   }
 
   // Used for testing TransformersCommand
   protected void setBundleContext(BundleContext bundleContext) {
     this.bundleContext = bundleContext;
   }
+
+  protected void setInputOption(Boolean b) {
```

Comment:
✏️ We should add the `@VisibleForTesting` annotation to these methods if they are for tests only.